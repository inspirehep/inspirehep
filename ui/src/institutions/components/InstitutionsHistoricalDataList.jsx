import React from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';
import InlineList, {
  SEPARATOR_MIDDLEDOT,
} from '../../common/components/InlineList';

function InstitutionsHistoricalDataList({ historicalData }) {
  return (
    <InlineList
      label="Note"
      items={historicalData}
      separator={SEPARATOR_MIDDLEDOT}
    />
  );
}

InstitutionsHistoricalDataList.propTypes = {
  historicalData: PropTypes.instanceOf(List),
};

export default InstitutionsHistoricalDataList;
