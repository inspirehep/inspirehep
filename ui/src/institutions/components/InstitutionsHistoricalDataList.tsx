import React from 'react';
import { List } from 'immutable';
import InlineList, {
  SEPARATOR_MIDDLEDOT,
} from '../../common/components/InlineList';

type Props = {
    historicalData?: $TSFixMe; // TODO: PropTypes.instanceOf(List)
};

function InstitutionsHistoricalDataList({ historicalData }: Props) {
  return (
    <InlineList
      label="Note"
      items={historicalData}
      separator={SEPARATOR_MIDDLEDOT}
    />
  );
}

export default InstitutionsHistoricalDataList;
