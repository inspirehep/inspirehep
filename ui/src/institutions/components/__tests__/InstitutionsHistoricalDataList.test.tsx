import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import InstitutionsHistoricalDataList from '../InstitutionsHistoricalDataList';

describe('InstitutionsHistoricalDataList', () => {
  it('renders', () => {
    const historicalData = fromJS(['This is my first note', 'second note']);
    const wrapper = shallow(
      <InstitutionsHistoricalDataList historicalData={historicalData} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
