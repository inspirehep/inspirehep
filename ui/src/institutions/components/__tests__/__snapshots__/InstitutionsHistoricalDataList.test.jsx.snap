// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstitutionsHistoricalDataList renders 1`] = `
<div
  className="__InlineList__"
>
  <span
    className=""
  >
    Note
    :
     
  </span>
  <ul>
    <li
      key="This is my first note"
    >
      This is my first note
       â€¢ 
    </li>
    <li
      key="second note"
    >
      second note
    </li>
  </ul>
</div>
`;
